py/server/deephaven/table_listener.py

```diff
@@ -87,9 +49,10 @@ def added(self, cols: Union[str, List[str]] = None) -> Dict[str, numpy.ndarray]:
             return {}
 
         try:
+            j_row_set = self.j_table_update.added.asRowSet()
             return next(
-                _changes_to_numpy(table=self.table, cols=cols, row_set=self.j_table_update.added.asRowSet(),
-                                  chunk_size=None))
+                _table_reader_chunk_dict(table=self.table, cols=cols, row_set= j_row_set,
+                                         chunk_size=j_row_set.size(), prev=False, to_numpy=True))
```

Comment:
`row_set.size()` is _not_ a good default, for the same reasons as discussed elsewhere.

Reply:
Adding a bit more context; in this case, the current interface already means to return a full numpy array. ie

```python
def added(self, cols: Union[str, List[str]] = None) -> Dict[str, numpy.ndarray]:
```

in this case, we shouldn't be doing any iteration, ie `io.deephaven.engine.rowset.RowSequence#getRowSequenceIterator` + `io.deephaven.engine.rowset.RowSequence.Iterator#getNextRowSequenceWithLength`; we should be able to use `j_row_set` directly with `_JTableUpdateDataReader.readChunkColumnMajor`.

Reply:
I actually believe reusing the generator interface here is a good trade off.

Reply:
BTW we are not defaulting to the `row_set.size()`. This is a call in which the interface is supposed to get all the `added` table update and return it to the listener.